app/src/main/java/org/oppia/android/app/translation/ActivityLanguageLocaleHandler.kt

```diff
@@ -0,0 +1,83 @@
+package org.oppia.android.app.translation
+
+import android.content.res.Configuration
+import android.util.Log
+import org.oppia.android.domain.locale.LocaleController
+import org.oppia.android.util.locale.OppiaLocale
+import javax.inject.Inject
+
+/**
+ * Activity-scoped handler for the current locale configured for all app layer components.
+ *
+ * Note that this handler acts as the single source of truth for the current display locale that
+ * should be used for all activity string formatting. The handler is automatically
+ * initialized by [AppLanguageLocaleHandler] with global app language and updated automatically if user
+ * changes language preferences and kept up-to-date with
+ * [AppLanguageWatcherMixin].
+ *
+ */
+class ActivityLanguageLocaleHandler @Inject constructor(
+  private val localeController: LocaleController,
+  private val appLanguageLocaleHandler: AppLanguageLocaleHandler
+) {
+  private lateinit var displayLocale: OppiaLocale.DisplayLocale
```

Comment:
This should be lazy initialized to the current display locale of ``AppLanguageLocaleHandler``.

Reply:
Thanks, updated this.